### PR TITLE
add dadjoke command.

### PR DESCRIPTION
**Pull request by savage4618**
This command uses an API (icanhazdadjoke) to pull in a random dadjoke. 

I had contacted the original creator of this code to get it re-done. They had confirmed to me it still worked and they used it on their self-host. I have tested the code on my own self-host now.

### CODE REVIEW

**Review by mrbentarikau — CHANGES_REQUESTED** [review]
Header,Set takes only two string arguments, this will error out on compile

File: stdcommands/dadjoke/dadjoke.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		req, err := http.NewRequest("GET", "https://icanhazdadjoke.com", nil)
+		if err != nil {
+			return nil , err
+		}
+		//Set the headers that will be sent to the API to determine the response.
+		req.Header.Set("Accept", "application/json")
+		req.Header.Add("User-Agent", "YAGPDB.xyz (https://github.com/botlabs-gg/yagpdb)")
+
+		client := &http.Client{}
+		apiResp, err := client.Do(req)
```
Comment (jo3-l) [inline stdcommands/dadjoke/dadjoke.go:36]:
```suggestion
		apiResp, err := http.DefaultClient.Do(req)
```

File: stdcommands/dadjoke/dadjoke.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		if err != nil {
+			return nil , err
+		}
+		//Once the rest of the function is done close our connection the API.
+		defer apiResp.Body.Close()
+		//Read the API response.
+		bytes, err := ioutil.ReadAll(apiResp.Body)
```
Comment (jo3-l) [inline stdcommands/dadjoke/dadjoke.go:43]: Package ioutil is deprecated. Use `io.ReadAll` instead.

File: stdcommands/dadjoke/dadjoke.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		if err != nil {
+			return nil , err
+		}
+		//Create our struct and unmarshal the content into it.
+		joke1 := Joke{}
```
Comment (jo3-l) [inline stdcommands/dadjoke/dadjoke.go:48]:
```suggestion
		var joke Joke
```

File: stdcommands/dadjoke/dadjoke.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		jsonErr := json.Unmarshal(bytes,&joke1)
```
Comment (jo3-l) [inline stdcommands/dadjoke/dadjoke.go:49]:
```suggestion
		err = json.Unmarshal(bytes,&joke1)
```

File: stdcommands/dadjoke/dadjoke.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		if jsonErr != nil {
+			return nil , jsonErr
+		}
```
Comment (jo3-l) [inline stdcommands/dadjoke/dadjoke.go:52]:
```suggestion
		if err != nil {
			return nil, err
		}
```

File: stdcommands/dadjoke/dadjoke.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+		//Return the joke - the other pieces are unneeded and ignored. 
+		resp := joke1.JOKE
+		return resp , nil
```
Comment (jo3-l) [inline stdcommands/dadjoke/dadjoke.go:55]:
```suggestion
		return joke.Joke, nil
```

**Review by jo3-l — COMMENTED** [review]
Some overarching suggestions:

- Run `gofmt` on this file.
- Many of the comments seem superfluous; consider removing.

**Comment by savage4618** [comment]
Thank you Joe for reviewing. I fixed all suggested lines and tested with success.

**Pull request merged**